portal-impl/src/com/liferay/portal/service/impl/UserServiceImpl.java

```diff
@@ -3556,26 +3550,35 @@ protected long[] checkOrganizations(long userId, long[] organizationIds)
 					organizationIds = ArrayUtil.append(
 						organizationIds, organization.getOrganizationId());
 				}
-
-				oldOrganizationIds[i] = organization.getOrganizationId();
 			}
 		}
 
 		// Check that the administrator has the permission to add a new
 		// organization and that the organization membership is allowed
 
+		if (!PropsValues.ORGANIZATIONS_ASSIGNMENT_STRICT ||
+			permissionChecker.isCompanyAdmin()) {
+
+			return organizationIds;
+		}
+
+		boolean allowed = false;
+
 		for (long organizationId : organizationIds) {
-			if ((oldOrganizationIds != null) &&
-				ArrayUtil.contains(oldOrganizationIds, organizationId)) {
+			if (OrganizationPermissionUtil.contains(
+					permissionChecker, organizationId,
+					ActionKeys.MANAGE_USERS)) {
 
-				continue;
-			}
+				allowed = true;
```

Comment:
I think it's easier to return here the `return organizationIds;` so you don't need the allowed variable

Reply:
I managed to do it in a simpler way, but we still need to iterate through all the organizations because the check needs to be done on all the organizations that the user1 is trying to add to the user2